lib/ssl_compat/ssl_compat.c

```diff
@@ -0,0 +1,109 @@
+#include <assert.h>
+#include <stddef.h>
+
+#include <openssl/ssl.h>
+
+#include "ssl_compat.h"
+
+#ifndef OPENSSL_VERSION_NUMBER
```

Comment:
Do we need this here any more?

Reply:
I can't think of a reason why we need this particular block.

We'll need some kind of LibreSSL compatibility check, because the hostname checking API was added in LibreSSL 2.7.0 (2018-03-21).  But that's only relevant to the 1.0.2 compatibility PR, not the 1.1.0 stage.